MemSQL Support in FlyWay

pradheeps opened the pull request:
Pull Request to address Issue [#1385](https://github.com/flyway/flyway/issues/1385)

Added the support for MemSQL. I have also included the MySQL Jdbc Driver along with the commit. MemSQL intentionally dropped certain MySQL features and hence needs separate implementation for it.
Also the flyway metadata table would be created as a MemSQL Reference table of row store type(will be kept in-memory) which would place this table in the master and all the leaf nodes.
In order to find out if it is a mysql or MemSQL connection, 
i get the catalog from the database and see if it contains the ‘memsql’ and decide if it is mysql or MemSQL as memsql has a built-in database named memsql
I have extended the MigrationTestCase and passed all except for the ones that are unsupported.
ConcurrentMigrationTest is not supported as SELECT FOR UPDATE like syntax is not supported in MemSQL.

This is my first open source contributions and i hope it goes well.

axelfontaine commented on flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/DbSupportFactory.java:
```diff
@@ -85,10 +89,16 @@ public static DbSupport createDbSupport(Connection connection, boolean printInfo
             //   ex.: Google SQL Service/MySQL
-            return new MySQLDbSupport(connection);
+            if(getCatalog(connection).contains("memsql")){
+                System.out.println("Instantiating MemSQL DbSuport");
```
This line should be removed

axelfontaine commented:
Thanks for the pull request.

Some remarks:

- Please remove the Jar from the source repository. It does not belong there.
- Look at my inline comments
- Could you add the autodetection to org.flywaydb.core.internal.util.jdbc.DriverDataSource?
- Could you add the connection URL template to flyway.conf?
- Could you submit a documentation PR for flywaydb.org including a database page, how to set up mysql for contributors as well as proper adjustments to the navigation in all relevant layouts and upding the database list on the homepage?
- Would you be willing to support this in the future if issues come up?

pradheeps commented:
- Please remove the Jar from the source repository. It does not belong there.

      Removed the jar folder and Driver

- Look at my inline comments

     Removed the print statement.

- Could you add the autodetection to org.flywaydb.core.internal.util.jdbc.DriverDataSource?

     Auto detection is already added and will be handled by the mysql section. There will be call to getcatalog method inside it and if the result has ‘memsql’ in it, then the MemSQL specific code will be used else mySQL 

- Could you add the connection URL template to flyway.conf?

    Added the url template for MemSQL which would be same as that of mySQL

- Could you submit a documentation PR for flywaydb.org including a database page, how to set up memsql for contributors as well as proper adjustments to the navigation in all relevant layouts and updating the database list on the homepage?

   https://github.com/flyway/flywaydb.org/pull/74

- Would you be willing to support this in the future if issues come up?

    Yes.

nycjay commented:
Any update on this PR?

nycjay commented:
@axelfontaine What is the likelihood of getting this PR accepted in an upcoming release.

apobbati commented:
Any updates on this PR? I can help with any cleanup tasks required to get it merged.

zirius666 commented:
Any updates on the PR it was moved to milestone 5.2 but that milestone was already released , Im a candidate to improve the missing things to add MemSql as other option of integration

mlantink commented:
Any update on this ? I do not see support in the latest 6.0 Flyway beta....is there still a planned release to add support ?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/flyway/flyway?pullRequest=1386) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/flyway/flyway?pullRequest=1386) before we can accept your contribution.<br/><hr/>**Pradheep Shanmugam** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/flyway/flyway?pullRequest=1386) it.</sub>

AdalbertMemSQL commented:
Some time ago, [MemSQL was renamed to SingleStore](https://www.singlestore.com/blog/revolution/).
Also, now it has its own [JDBC driver](https://docs.singlestore.com/db/v7.6/en/developer-resources/connect-with-application-development-tools/connect-with-java-jdbc/the-singlestore-jdbc-driver.html)
